src/coreclr/jit/valuenum.cpp

```diff
@@ -8742,7 +8654,17 @@ void Compiler::fgValueNumberTree(GenTree* tree)
             }
             else if (isVolatile)
             {
-                // For Volatile indirection, mutate GcHeap/ByrefExposed
+                // We just mutate GcHeap/ByrefExposed if isVolatile is true, and then do the read as normal.
```

Comment:
The comment is unfortunately not quite correct anymore, we don't actually do "read as normal" on this path; I will adjust it in the future VN changes (it is a very nice comment overall).